benchmarks/asv_delegated_conda.py

```diff
@@ -0,0 +1,201 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+ASV plug-in providing an alternative :class:`asv.plugins.conda.Conda`
+subclass that manages the Conda environment via custom user scripts.
+
+"""
+
+from os import environ
+from os.path import getmtime
+from pathlib import Path
+from shutil import copy2, copytree, rmtree
+from tempfile import TemporaryDirectory
+
+from asv import util as asv_util
+from asv.config import Config
+from asv.console import log
+from asv.plugins.conda import Conda
+from asv.repo import Repo
+
+
+class CondaDelegated(Conda):
+    """
+    Manage a Conda environment using custom user scripts, run at each commit.
+
+    Ignores user input variations - ``matrix`` / ``pythons`` /
+    ``conda_environment_file``, since environment is being managed outside ASV.
+
+    Original environment creation behaviour is inherited, but upon checking out
+    a commit the custom script(s) are run and the original environment is
+    replaced with a symlink to the custom environment. This arrangement is then
+    re-used in subsequent runs.
+
+    """
+
+    tool_name = "conda-delegated"
+
+    def __init__(
+        self,
+        conf: Config,
+        python: str,
+        requirements: dict,
+        tagged_env_vars: dict,
+    ) -> None:
+        """
+        Parameters
+        ----------
+        conf : Config instance
+
+        python : str
+            Version of Python.  Must be of the form "MAJOR.MINOR".
+
+        requirements : dict
+            Dictionary mapping a PyPI package name to a version
+            identifier string.
+
+        tagged_env_vars : dict
+            Environment variables, tagged for build vs. non-build
+
+        """
+        ignored = ["`python`"]
+        if requirements:
+            ignored.append("`requirements`")
+        if tagged_env_vars:
+            ignored.append("`tagged_env_vars`")
+        if conf.conda_environment_file:
+            ignored.append("`conda_environment_file`")
+        message = (
+            f"Ignoring ASV setting(s): {', '.join(ignored)}. Benchmark "
+            "environment management is delegated to third party script(s)."
+        )
+        log.warning(message)
+        requirements = {}
+        tagged_env_vars = {}
+        conf.conda_environment_file = None
+
+        super().__init__(conf, python, requirements, tagged_env_vars)
+        self._update_info()
+
+        self._env_commands = self._interpolate_commands(
+            conf.delegated_env_commands
+        )
+        self._delegated_env_parent = Path(
+            self._interpolate_commands(conf.delegated_env_parent)[0][0][0]
```

Comment:
`[0][0][0]`?! what is this object?

Reply:
Yeah that was way to obscure.

Any better? 16116c2c72c2a20068b8a2d3600e3fa8e716d24a